mountpoint-s3/tests/fuse_tests/mod.rs

```diff
@@ -263,3 +287,22 @@ pub fn read_dir_to_entry_names(read_dir_iter: ReadDir) -> Vec<String> {
         })
         .collect::<Vec<_>>()
 }
+
+#[macro_export]
+macro_rules! sleep_till_retry_succeed {
+    ($method_result:expr) => {
+        for i in 1..1000 {
+            match $method_result {
+                Ok(bool_value) => {
+                    if bool_value {
+                        break;
+                    } else {
+                        sleep(Duration::from_millis(10));
+                    }
+                }
+                Err(_) => panic!("The provided method could not provide result"),
+            }
+            assert!(i < 1000, "The provided method could not provide success result");
+        }
+    };
+}
```

Comment:
I'd try and make this more generic if possible.

Can we have this macro take a closure and a failure message?

I want to be able to write something like...

```rust
let test_condition = || test_client.contains_key(&format!("{}/{}", non_empty_dirname, filename)).unwrap();
wait_until!(test_condition, "timeout waiting for s3 to contain new object");
```

Reply:
Will need to learn how to do it, but yes I will do it.

Reply:
Working on it